info.go

```diff
@@ -259,8 +282,47 @@ func (pi *ProgramInfo) Instructions() (asm.Instructions, error) {
 		return nil, fmt.Errorf("unmarshaling instructions: %w", err)
 	}
 
-	// Tag the first instruction with the name of the program, if available.
-	insns[0] = insns[0].WithSymbol(pi.Name)
+	if pi.btf != 0 {
+		btfh, err := btf.NewHandleFromID(pi.btf)
```

Comment:
I just realised that this needs [`CAP_SYS_ADMIN`](https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L4475). That's a problem since newProgramInfoFromFd should be callable without SYS_ADMIN.

How about the following:
1. Populate `pi.funcInfos`, etc. as you currently do.
2. Move this block to `ProgramInfo.Instructions()`

That leaves the question what to do when `Instructions` is called without `CAP_SYS_ADMIN` and we get EPERM from `NewHandleFromID`?

Reply:
What you are describing is the current situation, is it not? The marked line which calls `btf.NewHandleFromID(pi.btf)` is inside `ProgramInfo.Instructions()` already. `newProgramInfoFromFd` can be called without `CAP_SYS_ADMIN`.

> That leaves the question what to do when Instructions is called without CAP_SYS_ADMIN and we get EPERM from NewHandleFromID?

Straight up changing could break current users. I think we can catch the `-EPERM` and fall back to old behavior. Then add to the docs that BTF info is only added when `CAP_SYS_ADMIN` is available. Would that be acceptable?

Reply:
Sorry, I didn't spot that in the diff :disappointed:

> Then add to the docs that BTF info is only added when CAP_SYS_ADMIN is available. Would that be acceptable?

Yeah that sounds okay.